### ISSUE TITLE
multiset of slice of sequence of subset-constrained sequence type gives type error

### ISSUE THREAD

**Issue by tjhance** [dafny-lang/dafny#353]
This code,
```
type A = s : seq<int> | |s| < 10

method f(a: seq<A>)
ensures multiset(a[..]) == multiset(a[..])
{
  
}
```
gives a type error,
```
$ dafny a.dfy
Dafny 2.3.0.10506
a.dfy(4,18): Error: type does not agree with element type of seq<A> (got ?)
a.dfy(4,37): Error: type does not agree with element type of seq<A> (got ?)
2 resolution/type errors detected in a.dfy
```

**Comment by RustanLeino** [comment]
I've confirmed that this is a problem in type inference. For now, here are two workarounds:

The simplest is to write `a` instead of `a[..]`, because those expressions refer to the same sequence.

But perhaps the repro you showed is distilled from a larger example, where you need something similar to the `[..]`. If so, if you introduce another variable to stand for the expression `a[..]`, you can workaround this bug. You can do this with a let expression, by replacing `multiset(a[..])` with `var u := a[..]; multiset(u)`.

**Comment by davidcok** [comment]
Fixed in PR #692, along with Issue #330
